BigQuery/tests/Unit/QueryJobConfigurationTest.php

```diff
@@ -206,4 +206,87 @@ public function parameterDataProvider()
             ]
         ];
     }
+
+    /**
+     * @dataProvider setParamTypesDataProvider
+     */
+    public function testSetParamTypes(array $values, array $types, array $expectedQuery)
+    {
+        $this->expectedConfig['configuration']['query'] = $expectedQuery
+        + $this->expectedConfig['configuration']['query'];
+
+        $this->config
+            ->parameters($values)
+            ->setParamTypes($types);
+
+        $this->assertEquals($this->expectedConfig, $this->config->toArray());
+    }
```

Comment:
Let's add some tests with error conditions (like those I've mentioned above), namely:

```php
$client->query($queryStr)->setParamTypes(['bar' => 'INT64']);
```
And
```php
// nonexistant named param
$client->query($queryStr)->parameters(['foo' => []])->setParamTypes(['bar' => 'INT64']);
// nonexistant ordered param
$client->query($queryStr)->parameters(['foo' => []])->setParamTypes(['INT64', 'STRING']);
```

Reply:
Thanks for the suggestions, resolved them all :)